src/mbed_tools/cli/build.py

```diff
@@ -76,7 +81,10 @@ def build(
        sterm: Open a serial terminal to the connected target.
        baudrate: Change the serial baud rate (ignored unless --sterm is also given).
     """
-    program = MbedProgram.from_existing(pathlib.Path(program_path))
+    if mbed_os_path is None:
```

Comment:
We shouldn't need this `if` statement, as we handle `mbed_os_path` being `None` in `MbedProgram::from_existing`

Reply:
> We shouldn't need this `if` statement, as we handle `mbed_os_path` being `None` in `MbedProgram::from_existing`

Actually, I'm slightly wrong here. We cast to a `Path` so we do need to check this isn't `None`.

Reply:
I had to change the function signature of `.from_existing` to accept a string instead of a Path to do that. `pathlib.Path(None)` does not return `None`.

let me know if that works for you :)

Reply:
I missed your second comment, i'll change it back 😂